### PR TITLE
Add tests for index ci to check if it functions correctly.

### PR DESCRIPTION
**Pull request by mohammedzee1000**

### CODE REVIEW

**Comment by bamachrn** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
@bamachrn @dharmit @rtnpro reviews please :)

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
Flaky epel causing flakes again :(

**Comment by mohammedzee1000** [comment]
#dotests

File: ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_01/test1.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    job-id: project1
+    git-url: https://github.com/mohammedzee1000/container-pipeline-service-indexci-testprojects
+    git-path: projects/baseproject/project1
+    git-branch: master
+    target-file: Dockerfile
+    desired-tag: latest
+    notify-email: mohammed.zee1000@gmail.com
+    depends-on: null
+
+  - id: 1
```
Comment (dharmit) [inline ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_01/test1.yml]: Both entries have `id: 1`? Is it on purpose?

File: ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_04/test1.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Projects:
+  - id: 1
+    app-id: test1
+    job-id: project1
+    git-path: projects/baseproject/project1
```
Comment (dharmit) [inline ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_04/test1.yml]: Is the `git-url` missing on purpose?

File: ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_01/test1.yml (hunk 1 above)
Comment (mohammedzee1000) [inline ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_01/test1.yml]: Yes, its if you check corresponding test case that uses this particular index, you should notice its checking for duplicate id

File: ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_04/test1.yml (hunk 2 above)
Comment (mohammedzee1000) [inline ci/tests/test_01_index_ci/test_01_malformed_index/test_index/test_04/test1.yml]: Yes, each test_xx corresponds to a test_xx_do_something function in the __init__ in same tests caegory directory. This one corresponds to https://github.com/CentOS/container-pipeline-service/pull/129/files#diff-d6f32f3093ffdb1abe1da59abed035eaR27 which checks for failure of index ci if git url is missing in index

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
@rtnpro review please

**Comment by navidshaikh** [comment]
This PR should merge before https://github.com/CentOS/container-pipeline-service/pull/156

**Pull request merged**